lib/ui/page/home/widget/scroll_keyboard_handler.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright © 2022-2025 IT ENGINEERING MANAGEMENT INC,
+//                       <https://github.com/team113>
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU Affero General Public License v3.0 as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License v3.0 for
+// more details.
+//
+// You should have received a copy of the GNU Affero General Public License v3.0
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/agpl-3.0.html>.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// Used in [_KeyRepeatHandler] in Timer to repeat
+const defaultRepeatPeriodDuration = Duration(milliseconds: 100);
+
+/// Default duration to scroll animation
+const defaultLongScrollAnimationDuration = Duration(milliseconds: 300);
+
+/// Default duration to scroll animation if PageUp/Down шы clamped
+const defaultQuickScrollAnimationDuration = Duration(milliseconds: 150);
+
+/// Factor to scrolling height
+const defaultScrollStepFactor = 0.9;
+
+/// [Widget] keyboard handler to PageUp(Option/Alt+Up), PageDown(Option/Alt+Down).
+/// Scroll [scrollController] to (accessible_height_from_constraints * ([scrollStepFactor] ?? 0.9)) in PageUp and PageDown
+class ScrollKeyboardHandler extends StatefulWidget {
+  const ScrollKeyboardHandler({
+    required this.scrollController,
+    required this.child,
+    this.scrollStepFactor = defaultScrollStepFactor,
```

Comment:
Does any widget change that parameter? If no, then there's no need to specify it there as a parameter.